core/embed/rust/src/ui/model_mercury/component/button.rs

```diff
@@ -527,44 +529,49 @@ impl IconText {
     ) {
         let width = self.text.map(|t| style.font.text_width(t));
 
-        let mut use_icon = false;
-        let mut use_text = false;
-
-        let mut icon_pos = Point::new(
-            area.top_left().x + ((Self::ICON_SPACE + Self::ICON_MARGIN) / 2),
-            area.center().y,
-        );
-        let mut text_pos = area.left_center() + baseline_offset;
-
         if area.width() > (Self::ICON_SPACE + Self::TEXT_MARGIN + width) {
-            //display both icon and text
-            text_pos = Point::new(area.top_left().x + Self::ICON_SPACE, text_pos.y);
-            use_text = true;
-            use_icon = true;
-        } else if area.width() > (width + Self::TEXT_MARGIN) {
-            use_text = true;
-        } else {
-            //if we can't fit the text, retreat to centering the icon
-            icon_pos = area.center();
-            use_icon = true;
-        }
-
-        if use_text {
             self.text.map(|t| {
+                let mut text_pos = area.left_center() + baseline_offset;
+                text_pos = Point::new(area.top_left().x + Self::ICON_SPACE, text_pos.y);
                 shape::Text::new(text_pos, t)
                     .with_font(style.font)
                     .with_fg(style.text_color)
                     .with_alpha(alpha)
                     .render(target)
             });
+        } else {
+            self.text.map(|t| {
+                let mut rects: Vec<Rect, 2> = Vec::new();
+                let _ = rects.push(Rect::new(area.top_left(), area.right_center()));
+                let _ = rects.push(Rect::new(area.left_center(), area.bottom_right()));
+                let mut texts: Vec<&str, 2> = Vec::new();
+                let (t1, t2) = split_two_lines(
+                    t,
+                    style.font,
+                    area.width() - Self::ICON_SPACE - Self::TEXT_MARGIN,
+                );
+                let _ = texts.push(t1);
+                let _ = texts.push(t2);
+                for i in 0..texts.len() {
+                    let mut text_pos = rects[i].left_center() + baseline_offset;
+                    text_pos = Point::new(rects[i].top_left().x + Self::ICON_SPACE, text_pos.y);
```

Comment:
style: what you want is not a `mut` variable but a rebind:
```suggestion
                    let text_pos = rects[i].left_center() + baseline_offset;
                    let text_pos = Point::new(rects[i].top_left().x + Self::ICON_SPACE, text_pos.y);
```